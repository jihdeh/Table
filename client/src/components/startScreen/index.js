import React from 'react';
import JoinButton from './joinButton';

export default () => (
  <div  className="App">
    <JoinButton />
  </div>
);
