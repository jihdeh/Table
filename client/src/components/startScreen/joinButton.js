import React from 'react';
import res from '../../resources/pages/startScreen';
import { goToPath } from '../../helpers/routing';
import paths from '../../paths';

export default () => (
  <button
    onClick={ () => goToPath(paths.play) }
    className="player-join-button"
    >
    { res.playerJoinButton }
  </button>
);
