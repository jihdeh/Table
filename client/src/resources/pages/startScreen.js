export default {
  playerJoinButton: 'Join as player',
  tableJoinButton: 'Join as table'
};
