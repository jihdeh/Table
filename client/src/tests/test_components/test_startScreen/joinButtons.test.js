import { PlayerJoinButton, TableJoinButton } from '../../../components/startScreen/joinButtons';

import React from 'react';

import ReactTestUtils from 'react-addons-test-utils';
const shallowRenderer = ReactTestUtils.createRenderer();

it('PlayerJoinButton renders as before', () => {
  const component = shallowRenderer.render(<PlayerJoinButton />);
  expect(component).toMatchSnapshot();
});

it('TableJoinButton renders as before', () => {
  const component = shallowRenderer.render(<TableJoinButton />);
  expect(component).toMatchSnapshot();
});
