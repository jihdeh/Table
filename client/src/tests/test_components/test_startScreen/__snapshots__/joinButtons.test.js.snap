exports[`test PlayerJoinButton renders as before 1`] = `
<button
  className="player-join-button"
  onClick={[Function]}>
  Join as player
</button>
`;

exports[`test TableJoinButton renders as before 1`] = `
<button
  className="table-join-button"
  onClick={[Function]}>
  Join as table
</button>
`;
